### PR TITLE
PMax Banner Reports Section

### PR DESCRIPTION
**Pull request by jorgemd24**
### Changes proposed in this Pull Request:

This PR adds the banner in the GLA Reports Section with information related to the PMax migration status. See more context in the following links:

- [Design Proposal i2: Performance Max Migration (Milestone 2)](https://woodesigngrowcontractor.wordpress.com/2022/04/20/design-proposal-i2-performance-max-migration-milestone-2/)
- [Figma File](https://href.li/?https://www.figma.com/file/eQ9O4m2flzrcAiDMXkOW0m/Google-Listings-%26-Ads-v1.x?node-id=7840%3A127644)
- https://github.com/woocommerce/google-listings-and-ads/pull/1472

### Before Migration

- The banner should not be displayed.

### After Migration

- If the user has been converted, show the following notice:

image

- Banner goes away when X’d, or after 1 month after it is first displayed.

### Detailed test instructions:
The conversion status is determined for the wp option: `gla_campaign_convert_status`, it is an object with two properties: status & updated. The property updated is a UNIX time in seconds that indicates when was the last time that the status was checked until it becomes converted.

#### Unconverted
a:2:{s:6:"status";s:11:"unconverted";s:7:"updated";i:1651517855;} //May 02 2022

#### Converted
a:2:{s:6:"status";s:9:"converted";s:7:"updated";i:1651517855;} //May 02 2022

If you have Smart Shopping campaigns, the starting status will be 'unconverted', if all campaigns are PMax then your status will be 'converted'.

### Before Migration

1. Go to the GLA Reports
2. The banner should not appear.

### After Migration

1. Update manually the wp option to: `a:2:{s:6:"status";s:9:"converted";s:7:"updated";i:1651517855;}` //May 02 2022
2. Go to the GLA Reports
3. Check that the notice is showing

image

4. If you click X, the banner disappears. Remove from the localStorage the item `gla-is-reports-migration-notice-dismissed` if you like to see it again.
5. Change manually the time in the wp option to be > 1 month ago and the banner should be gone. Example: `a:2:{s:6:"status";s:9:"converted";s:7:"updated";i:1646308364;}` //Mar 03 2022

### Additional details:

The link for "Learn more about this upgrade" is pending to be confirmed.

### Changelog entry

> Add - Pmax migration banner reports

### CODE REVIEW

File: js/src/components/campaign-conversion-notice/reports-notice.js
Comment (ianlin) [inline js/src/components/campaign-conversion-notice/reports-notice.js]: This component was actually very similar to the component `js/src/components/campaign-conversion-notice/index.js` that implemented in the previous PR, I'm thinking maybe we could add a shared component and accept props like `localStorageKey`, `className`, and `children` etc so we could reuse some codes.

File: js/src/components/campaign-conversion-notice/reports-notice.js
Comment (ianlin) [inline js/src/components/campaign-conversion-notice/reports-notice.js]: I think the event name of `AppDocumentationLink` should always be `gla_documentation_link_click` that is defined in `AppDocumentationLink` component. When we want to distinguish different link clicks we can use the event properties, in this case `linkId` or something else. So an analyst can get the stats of the total document link clicks, as well as clicks per link by filtering out the event properties. It might need to be changed in `js/src/components/campaign-conversion-notice/index.js` from the previous PR.

However from the entire code base there were some places that defined the event name for `AppDocumentationLink`, I will create a follow-up issue for it later.

File: js/src/components/campaign-conversion-notice/index.scss
Comment (ianlin) [inline js/src/components/campaign-conversion-notice/index.scss]: 💅  It could be simplified to `margin: $grid-unit-40 0 $grid-unit-10;`

File: js/src/components/campaign-conversion-notice/index.scss
Comment (ianlin) [inline js/src/components/campaign-conversion-notice/index.scss]: 💅  From the design the left and right padding is just 20px, so maybe we can change the left and right padding to `$grid-unit-30`.

**Review by ianlin — COMMENTED** [review]
Thanks for adding this, tested locally and the behaviour was working as expected. I left some comments that might be worth discussing.

File: js/src/components/campaign-conversion-notice/index.scss
Comment (jorgemd24) [reply js/src/components/campaign-conversion-notice/index.scss]: Added [d6b5e5e](https://github.com/woocommerce/google-listings-and-ads/pull/1492/commits/d6b5e5e45a41fea9cbcc92691145da9adb16f4cb)

File: js/src/components/campaign-conversion-notice/index.scss
Comment (jorgemd24) [reply js/src/components/campaign-conversion-notice/index.scss]: Added [c832e74](https://github.com/woocommerce/google-listings-and-ads/pull/1492/commits/c832e746f984f81da8c9945fd00ba5ac599b4a6d)

File: js/src/components/campaign-conversion-notice/reports-notice.js
Comment (jorgemd24) [reply js/src/components/campaign-conversion-notice/reports-notice.js]: Thanks, Ian for the explanation! I remove the eventName from AppDocumentationLink. See [d214109](https://github.com/woocommerce/google-listings-and-ads/pull/1492/commits/d214109c486ee8369f27f951e9b24476e1f29784)

**Comment by jorgemd24** [comment]
Thanks, @ianlin for reviewing this and your comments! I have updated the code with your suggestions and I refactor the previous banner and the reports banner so we can reuse more code. 

This PR is ready for another round!

File: js/src/components/campaign-conversion-notice/migration-notice.js
Comment (ianlin) [inline js/src/components/campaign-conversion-notice/migration-notice.js]: 💅 Just a typo, should be "conversion".

File: js/src/components/campaign-conversion-notice/index.js
Comment (ianlin) [inline js/src/components/campaign-conversion-notice/index.js:35]: I couldn't comment on the lines that are not part of the changes, but in the events JSDoc above this, we are no longer sending the event name `gla_upgrade_campaign_learn_more_link_click` so the comment can be removed.

File: js/src/components/campaign-conversion-notice/reports-notice.js
Comment (ianlin) [inline js/src/components/campaign-conversion-notice/reports-notice.js]: We are no longer sending the event name `gla_upgrade_campaign_learn_more_link_click` so the comment can be removed.

File: js/src/components/campaign-conversion-notice/conversion-statuses.js
Comment (ianlin) [inline js/src/components/campaign-conversion-notice/conversion-statuses.js]: 💅  The string doesn't need to be wrapped in `{}`.

File: js/src/components/campaign-conversion-notice/migration-notice.js
Comment (ianlin) [inline js/src/components/campaign-conversion-notice/migration-notice.js]: I feel like this component could be promoted to a standard shared component in this project, if we name it `DismissibleNotice` and put it under `js/src/components`, and only set the local storage if the provided prop `localStorageKey` is not empty.

But it's not a blocking part for this PR, we can do it later when doing other TODOs in the future. E.g. when the "learn more" link is ready

File: js/src/components/campaign-conversion-notice/index.js
Comment (ianlin) [inline js/src/components/campaign-conversion-notice/index.js:15]: ❔ Question for importing scss: I'm not familiar with css/scss, but we import `index.scss` in this file but did not import it in `campaign-conversion-notice/reports-notice.js`. The styles of the report notice looks great on the browser so it means there is no problem as the styles are loaded by `campaign-conversion-notice/index.js`. I feel like it's not a good way as it's a bit implicit. So maybe we could move the styles of `reports-notice.js` to, for example, `reports-notice.scss` and import it from `reports-notice.js`?

File: js/src/components/campaign-conversion-notice/report-notice.test.js
Comment (ianlin) [inline js/src/components/campaign-conversion-notice/report-notice.test.js]: 💅  This test file's name is `report-notice.test.js` while the filename of the component it's testing on is `reports-notice.js`.

**Review by ianlin — APPROVED** [review]
Hey @jorgemd24, thanks for all the changes. I approve it in advance but still left some suggestions in the comments. The only comment that I think should be changed before merging is to remove the event JSDoc for `gla_upgrade_campaign_learn_more_link_click`.

File: js/src/components/campaign-conversion-notice/index.js
Comment (jorgemd24) [reply js/src/components/campaign-conversion-notice/index.js:15]: Yeah thats a good idea, I have changed in [ebedc71](https://github.com/woocommerce/google-listings-and-ads/pull/1492/commits/ebedc71cd053261e42eb4f2ce3b74b3fe2bdba86)

File: js/src/components/campaign-conversion-notice/migration-notice.js
Comment (jorgemd24) [reply js/src/components/campaign-conversion-notice/migration-notice.js]: I had the same feeling! I made some changes to the component, basically:
- I updated the name.
- It checks if the `localStorageKey ` is set or not.

However, I think if we want to leave it as an independent component, it would be convenient to add some callback function on the onRemove function so it is more customized depending on the user's needs. I think for now I will leave it as it is and in a different PR, we can focus to make this component a standard shared component.

**Comment by jorgemd24** [comment]
Thanks, @ianlin for the second review! I have added your suggestions and I left a comment about making the `DismissableNotice` a shared component.

File: js/src/components/campaign-conversion-notice/migration-notice.js
Comment (ianlin) [reply js/src/components/campaign-conversion-notice/migration-notice.js]: Makes sense, thanks for the explanation!

**Review by ianlin — APPROVED** [review]
Thanks for all the changes, it's good to go!

**Pull request merged**
